Source/Serilog.Exceptions/Core/ExceptionPropertiesBag.cs

```diff
@@ -56,10 +63,46 @@ public void AddProperty(string key, object? value)
                 }
             }
 
-            this.properties.Add(key, value);
+            this.AddPairToProperties(key, value);
         }
 
         /// <inheritdoc />
         public bool ContainsProperty(string key) => this.properties.ContainsKey(key);
+
+        /// <summary>
+        /// We want to be as robust as possible
+        /// so even in case of multiple properties with the same name
+        /// we want to at least try carrying on and keep working.
+        /// </summary>
+        private void AddPairToProperties(string key, object? value)
+        {
+#if NET5_0
+            var i = 0;
+            while (!this.properties.TryAdd(key, value) && i < AcceptableNumberOfSameNameProperties)
+            {
+                key += "$";
+                i++;
+            }
+#else
+            key = this.MakeSureKeyIsUnique(key);
+
+            this.properties.Add(key, value);
+#endif
+        }
+
+#if !NET5_0
+        
+        private string MakeSureKeyIsUnique(string key)
+        {
+            var i = 0;
+            while (this.properties.ContainsKey(key) && i < AcceptableNumberOfSameNameProperties)
+            {
+                key += "$";

```

Comment:
Perhaps it would be better to prefix the key with the name of the class the property came from instead e.g.

```
BaseClass.Foo
DerivedClass.Foo
```

This would make the name longer but this is a rare case and it would make it easier to understand which duplicate property came from where.

Reply:
That makes sense, agreed, but the `ExceptionPropertiesBag` operates on `string key` already, without the context of where it came from.

In the second PR, we may discuss whether to use that convention or just discard `BaseClass.Foo`.